src/DotLiquid/Tags/AddFilters.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text.RegularExpressions;
+using DotLiquid.Exceptions;
+using DotLiquid.Util;
+
+namespace DotLiquid.Tags
+{
+    /// <summary>
+    /// Adds filters in a whitelisted class to the current context.
+    ///
+    /// {%- addfilters 'ShopifyFilters' -%}
+    /// </summary>
+    public class AddFilters : Tag
+    {
+        private static readonly Regex Syntax = R.B(R.Q(@"\s*([""'].+[""'])\s*"));
+
+        private static readonly IDictionary<string, Type> FILTER_WHITELIST = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+        static AddFilters()
+        {
+            WhitelistFilter(typeof(ShopifyFilters));
+        }
+
+        /// <summary>
+        /// Add the provided class to the whitelist of classes that can be added by a Liquid Designer
+        /// </summary>
+        /// <param name="filterClassType">A class containing filter operations.</param>
+        /// <param name="alias">An alias for the class, if not provided the classes short name is used</param>
+        public static void WhitelistFilter(Type filterClassType, string alias = null)
+        {
+            alias = alias ?? filterClassType.Name;
+            FILTER_WHITELIST[alias] = filterClassType;
+        }
+
+        private string alias;
+
+        /// <inheritdoc/>
+        public override void Initialize(string tagName, string markup, List<string> tokens)
+        {
+            Match syntaxMatch = Syntax.Match(markup);
+            if (!syntaxMatch.Success)
+                throw new SyntaxException(Liquid.ResourceManager.GetString("AddFiltersTagSyntaxException"));
+
+            alias = syntaxMatch.Groups[1].Value;
+
+            base.Initialize(tagName, markup, tokens);
+        }
+
+        /// <inheritdoc/>
+        public override void Render(Context context, TextWriter result)
+        {
+            var renderedAlias = getAliasValue(context);
```

Comment:
What's wrong with context[alias] ? This is how include tag works - context handles quotes as literal and without as a variable. The difference, off the top of my head, is that new Variable will allow you to specify a filter as in 
```
{% addfilters unknown_variable | default: 'ShopifyFilters' %} 
```
which I don't think you were trying to do

Reply:
I based AddFilters on Assign, which uses new Variable. I've simplified it now to use context[alias].toString()